app/js/components/App.jsx

```diff
@@ -6,15 +6,76 @@
  * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
  * graphic logo is a trademark of OpenMRS Inc.
  */
-import React from 'react';
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { ToastContainer, toast } from 'react-toastify';
 import Dashboard from './Dashboard';
 import Header from './header/Header';
+import axiosInstance from '../config/axiosInstance';
 
-const App = () => (
-  <div>
-    <Header />
-    <Dashboard />
-  </div>
-);
+export default class App extends Component {
+  constructor(props) {
+    super(props);
 
-export default App;
+    this.state = {
+      authenticated: false,
+    };
+  }
+
+  componentDidMount() {
+    this.checkAuthentication();
+  }
+
+  checkAuthentication = () => {
+    this.getLoginStatus()
+      .then(response => this.handleResponse(response))
+      .catch(error => this.handleError(error));
+  }
+
+  getLoginStatus = () => (
+    new Promise((resolve, reject) => {
+      axiosInstance.get('/session')
+        .then(response => resolve(response))
+        .catch(error => reject(error));
+    })
+  )
+
+  handleResponse = (response) => {
+    response.data.authenticated
+      ? this.setState({ authenticated: true })
+      : this.props.history.push('/login');
+  }
+
+  handleError = (error) => {
+    if (error.response) {
+      this.toastError(error.response.data.error.message);
+      setTimeout(() => {
```

Comment:
Great job. However, I see you used setTimeout with a delay of 3 seconds. Why did you use setTimeout? Isn't a delay of 3 seconds to much? I think the error notification can display in the login page without waiting for 3 seconds.

Reply:
Yes @emmabaye, great observation. This is what that does:
When a user tries to access the root route component "/", a loader is displayed while the authentication API call happens and in case there is a server error response, that is displayed to the user for 3 seconds before redirecting to the login page. However, this ticket: https://issues.openmrs.org/browse/PLAT-57 will handle the creation of a single toastr to handle all application notifications and the 3 seconds delay before redirecting will no longer be necessary. I hope this provides some context.

Reply:
If this is going to be fixed, then why not just remove it?